### ISSUE TITLE
zh-CN and zh-Hant locales no longer work 

### ISSUE THREAD

**Issue by rebornix** [Microsoft/vscode#53275]
Issue Type: <b>Bug</b>

Changing the display language to `zh-CN` or `zh-Hant` doesn't show display language in Chinese and no prompt for language pack installation suggestion. 

`zh-CN` and `zh-Hant` works in latest Stable.

VS Code version: Code - Insiders 1.25.0-insider (07d89abf1522a67f0e8105770505077f454ab021, 2018-06-28T05:19:16.448Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz (8 x 3600)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: disabled_software<br>video_decode: enabled<br>video_encode: enabled<br>vpx_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|31.92GB (21.20GB free)|
|Process Argv|C:\Program Files\Microsoft VS Code Insiders\Code - Insiders.exe --enable-proposed-api Microsoft.vscode-pull-request-github|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (6)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-eslint|dba|1.4.12
gitlens|eam|8.4.1
tslint|eg2|1.0.33
vscode-pull-request-github|Mic|0.0.3
theme-monokai-pro-vscode|mon|1.1.8
debugger-for-chrome|msj|4.7.0

(1 theme extensions excluded)

</details>

**Comment by dbaeumer** [comment]
@ramya-rao-a just to make sure this will work correctly in all cases:

- there is no problem of having `zh-hans` or `zh-hant` in the extension name
- the language id must always be `zh-cn` or `zh-tw`. The ids are API (extension might rely on it) and we can't change them. These are also the value returned from the Electron OS call to get the app locale.

This means that these two statements

```
const languageName = loc ? (loc.languageName || currentLocale) : currentLocale;
const languageDisplayName = loc ? (loc.localizedLanguageName || loc.languageName || currentLocale) : currentLocale;
```

are OK iff the language pack defined a id. If not (which we could argue should not happen) the code might not work since we should use the locale and not the currentLocale.

**Comment by sandy081** [comment]
@ramya-rao-a This broke the recommendation prompts as the change is using the mapped locale id for getting minimal translations. 

As @dbaeumer mentioned, we should use `zh-hans` or `zh-hant` only for extension names and locales should be always `zh-cn` and `zh-tw`. Pushed a fix for this here -3962b545e89b373168ba5d7ff08b7e340879ef4b
